apps/blas/axpy/axpy.fil

```diff
@@ -0,0 +1,41 @@
+import "apps/blas/scal/scal.fil";
+import "apps/blas/util.fil";
+
+// Performs a*x + y, where x,y are vectors and a is a scalar
+// W: Width of nums
+// N: Length of vectors
+// M: Number of multipliers
+// A: Number of adders
+comp Axpy[W, N, M, A]<'G:II>(
+  go:   interface['G],
+  a:    ['G, 'G+1] W,
+  x[N]: ['G, 'G+1] W,
+  y[N]: ['G, 'G+1] W,
+) -> (
+  out[N]: ['G+L, 'G+L+1] W
+) with {
+  some L where L > 0;
+  some II where II > 0;
+} where W > 0,
+        L > 0,
+        N > 0,
+        M > 0,
+        A > 0
+{
+  // Perform a*x
+  Sc := new Scal[W, N, M];
+  s := Sc<'G>(x{0..N}, a);
+
+  // register y
+  y_reg := new Shift[W, Sc::L, N]<'G>(y{0..N});
```

Comment:
Interesting! There is a different version of this that "fuses" the multiplication and adding so that we don't need to wait for all of `a*x` to be computed before computting `a*x+y`. Thoughts on how hard that would be express? I almost want to say that `Scal` (I'm assuming scalar product) should be implemented in terms of axpy.

Reply:
Ooh, interesting! I will try that, it sounds like it shouldn't be too hard to express!